os/watch/src/package.scala

```diff
@@ -45,4 +52,6 @@ package object watch{
     thread.start()
     watcher
   }
+
+  var use_linux_inotify: Boolean = false
```

Comment:
I don't like the idea of switching this globally. I'd prefer a parameter which enables "native" support. The scaladoc should explain what "native" means for each platform. Also, making this binary compatible would be good idea, e.g. keeping the old `def watch` signature with sensible default to current behavior.

There is one issue though, as for Mac OS there seems to be be currently no choice but only the native implementation. I can't tell if it's because the native version just works better or because the generic version doesn't work at all.

Reply:
I did it this way in order to stress the fact that it's an internal implementation detail not intended to survive for a long time and should be neither documented not show up on the public API.

A reasonable alternative is to add an extra argument to `os.watch`. Naming it something like `preferNative` would suggest that it's just a preference. It will always be picked on MacOS because it's the only choice.

Reply:
https://github.com/com-lihaoyi/os-lib/pull/70/commits/3760863210ff8d9af4466d4d8b2c70e1da2eb6e2

Reply:
@lefou the generic OSX version uses polling on every watched file, which is as good as "doesn't work at all" on non-trivial folder hierarchies

Reply:
I like the idea with the `WatchConfig`, as it avoids future changes to `watch` API for such kind of tweaks. I don't like the `deprecated` annotation without any hint though. Also, we should give the user the fine control she wants, and make a flag per platform, when appropriate. And after @lihaoyi s comment I think it will be end up with only one flag right now: `useLinuxInotify`.

Reply:
I think when we make the new `NotifyWatcher` optional and disabled by default, this is good enough to merge as-is. I doubt, we will find a reviewer familiar enough with Linux inotify system to review this timely.